### PR TITLE
Explaining/Fixing asyncio allowlist exceptions

### PR DESCRIPTION
**Pull request by hatal175**
Went over asyncio allowlist exceptions, and found out why they were there. Fixed where I could and added explanation where I couldn't

### CODE REVIEW

File: tests/stubtest_whitelists/py3_common.txt [hunk 1]
```diff
@@ -35,22 +35,18 @@ abc.abstractclassmethod
...
+# Condition functions are exported in __init__
 asyncio.locks.Condition.acquire
 asyncio.locks.Condition.locked
 asyncio.locks.Condition.release
-asyncio.proactor_events.BaseProactorEventLoop.sock_recv
-asyncio.selector_events.BaseSelectorEventLoop.sock_recv
-asyncio.streams.FlowControlMixin.__init__
-asyncio.tasks.Task.get_stack
-asyncio.tasks.Task.print_stack
+asyncio.proactor_events.BaseProactorEventLoop.sock_recv # nbytes parameter has different name 'n' in implementation
```
Comment (JelleZijlstra) [inline tests/stubtest_whitelists/py3_common.txt:48]: Why can't we fix this in the stubs?

**Review by JelleZijlstra — COMMENTED** [review]
Thanks, this is very useful! I'll do a full review later, just something quick I noticed.

**Comment by hatal175** [comment]
It's different in the actual python source code rather than typeshed.
Can this be fixed in the stubs? I assumed not, but I can try?

**Comment by JelleZijlstra** [comment]
The stubs should reflect what the source does, unless there's something I'm missing. Do you mean it's different in the docs perhaps?

**Comment by hatal175** [comment]
In BaseProactorEventLoop
async def sock_recv(self, sock, n):

In AbstractEventLoop
async def sock_recv(self, sock, nbytes):
raise NotImplementedError

**Comment by JelleZijlstra** [comment]
I see, that makes sense. Ideally that should be fixed in CPython, maybe by making them positional-only, but for typeshed it makes sense to just leave the allowlist entry.

File: stdlib/asyncio/base_events.pyi [hunk 2]
```diff
@@ -184,7 +200,7 @@ class BaseEventLoop(AbstractEventLoop, metaclass=ABCMeta):
         ) -> _TransProtPair: ...
     if sys.version_info >= (3, 7):
         async def sock_sendfile(
-            self, sock: socket, file: IO[bytes], offset: int = ..., count: Optional[int] = ..., *, fallback: bool = ...
+            self, sock: socket, file: IO[bytes], offset: int = ..., count: Optional[int] = ..., *, fallback: Optional[bool] = ...
```
Comment (JelleZijlstra) [inline stdlib/asyncio/base_events.pyi:203]: Why do you need this change? In the implementation I don't see any indication `None` is accepted; the default value is `True`.

Comment (JelleZijlstra) [reply stdlib/asyncio/base_events.pyi:203]: Oh it defaults to None in events.py. Seems like a mistake in the code, but the Optional here is harmless, so let's not worry about it.

**Comment by hauntsaninja** [comment]
Thanks, this is great!

**Pull request merged**